src/plugins/external_tool_support/src/blast_plus/AlignToReferenceBlastDialog.cpp

```diff
@@ -96,15 +96,19 @@ AlignToReferenceBlastCmdlineTask::AlignToReferenceBlastCmdlineTask(const Setting
       settings(settings),
       cmdlineTask(NULL),
       loadRef(NULL),
-      reportFile(AppContext::getAppSettings()->getUserAppsSettings()->getCurrentProcessTemporaryDirPath() + "/align_to_ref_XXXXXX.txt") {
+      reportFile(AppContext::getAppSettings()->getUserAppsSettings()->getCurrentProcessTemporaryDirPath() + "/align_to_ref_XXXXXX.txt"),
+      errorStateFile(AppContext::getAppSettings()->getUserAppsSettings()->getCurrentProcessTemporaryDirPath() + "/align_to_ref_error_state_XXXXXX.txt") {
     GCOUNTER(cvar, "AlignToReferenceBlastCmdlineTask");
 }
 
 void AlignToReferenceBlastCmdlineTask::prepare() {
     AppContext::getAppSettings()->getUserAppsSettings()->createCurrentProcessTemporarySubDir(stateInfo);
-    const bool opened = reportFile.open();
-    SAFE_POINT_EXT(opened, setError(L10N::errorOpeningFileWrite(reportFile.fileName())), );
-    reportFile.close();
+    QList<QTemporaryFile*> tempFiles = { &reportFile, &errorStateFile };
+    for (auto tempFile : tempFiles) {
+        bool opened = tempFile->open();
```

Comment:
nit: A good candidate for and utility method like: FileAndDirectoryUtils::checkFileIsWritable(QFile&, OpStatus&);

Reply:
OK